hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SpecificPartitionsImapReaderPmsTest.java

```diff
@@ -22,110 +22,233 @@
+                .getJobCoordinationService();
+
+        jobInvocationObserver = new JobInvocationObserverImpl();
+        jobCoordinationService.registerInvocationObserver(jobInvocationObserver);
+
+        perMemberOwnedPKey = new int[MEMBERS - 1];
+        perMemberOwnedPId = new int[MEMBERS - 1];
+
+        Address coordinatorAddress = instance().getCluster().getLocalMember().getAddress();
         for (int i = 1; i < ITERATIONS; ++i) {
             int pIdCandidate = instance().getPartitionService().getPartition(i).getPartitionId();
-            if (Arrays.binarySearch(partitionAssignment.get(ownderAddress), pIdCandidate) >= 0) {
-                pKey = i;
-                keyOwnerPartitionId = pIdCandidate;
+            if (reversedPartitionAssignment.get(pIdCandidate).equals(coordinatorAddress)) {
+                coordinatorOwnedPartitionKey = i;
+                coordinatorOwnedPartitionId = pIdCandidate;
                 break;
             }
         }
+
+        Set<Address> clusterAddresses = new HashSet<>(partitionAssignment.keySet());
+        clusterAddresses.remove(coordinatorAddress);
+        int j = 0;
+        for (int i = 1; i < ITERATIONS && !clusterAddresses.isEmpty(); ++i) {
+            int pIdCandidate = getNodeEngineImpl(instance()).getPartitionService().getPartitionId(i);
+            Address address = reversedPartitionAssignment.get(pIdCandidate);
+            if (clusterAddresses.contains(address)) {
+                perMemberOwnedPKey[j] = i;
+                perMemberOwnedPId[j++] = pIdCandidate;
+                clusterAddresses.remove(address);
+            }
+        }
     }
 
     // Note: basic functionality must work even without member pruning used.
     @Test
     public void test_basic() {
         // Basic test is performed as submitting job by DAG.
-        IMap<Integer, Integer> map = instance().getMap(mapName);
-        map.put(0, 0);
+        sourceMap.put(0, 0);
 
         DAG dag = new DAG();
-        ProcessorMetaSupplier readPms = mapReader(mapName, null, List.of(
-                List.of(ConstantExpression.create(0, QueryDataType.INT))));
+        ProcessorMetaSupplier readPms = mapReader(mapName, null, null);
         Vertex source = dag.newVertex("source", readPms);
         Vertex sink = dag.newVertex("sink", writeMapP(sinkName));
 
         dag.edge(between(source, sink));
         instance().getJet().newLightJob(dag, jobConfig).join();
 
         assertEquals(0, instance().getMap(sinkName).get(0));
+
+        Vertex sourceV = dag.getVertex("source");
+        assertNotNull(sourceV);
+
+        ProcessorMetaSupplier metaSupplier = sourceV.getMetaSupplier();
+        assertNotNull(metaSupplier);
+        assertInstanceOf(SpecificPartitionsImapReaderPms.class, metaSupplier);
+
+        SpecificPartitionsImapReaderPms pms = (SpecificPartitionsImapReaderPms) metaSupplier;
+        // Ensure we scan all partitions.
+        assertNull(pms.partitionsToScan);
```

Comment:
??? is that unit test of `mapReader` static method? If so, it should be separate. PMS is immutable in DAG - do you expect that something might have changed it?

Reply:
ok, you rely on the fact that DAG is not serialized when submitting light job (which is only optmiziation implementation detail). Still, this feels more like unit test. Mixing layers and depending on implementation deatls makes the higher level test less readable.

Reply:
> Mixing layers and depending on implementation details makes the higher level test less readable.

I do not mix anything here, I am working on `CoreAPI` to have access to vertexes, PMS-es to check all I need.
Again, tt was done **specially** on CoreAPI level with purpose. E.g, take a look on `WriteFilePTest`

Reply:
you mix integration test with unit test. I meant this (method names are not perfect): https://github.com/hazelcast/hazelcast/pull/25206/commits/32f13586ee34c8c89ffb8cd80d5f25bede4c2c8f

Reply:
it would be nice to even even more unit tests, the algorithm in SpecificPartitionsImapReaderPms is quite complicated and not all cases and branches are covered, we also lack unit tests with non-null strategy